test/github_changelog/conventional_test.clj

```diff
@@ -1,78 +1,92 @@
+          pull (sample-pull {:body body})]
       (is (= [["JIRA-1" jira-issue-url]] (sut/parse-issues config pull)))))
   (testing "with a GitHub issue"
-    (let [pull (generate-pull "Fixes #1")]
-      (is (= [["#1" (str repo-url "/issues/1")]] (sut/parse-issues config pull))))))
+    (let [pull (sample-pull {:body "Fixes #1"})]
+      (is (= [["#1" (str github-url "/issues/1")]] (sut/parse-issues config pull))))))
 
-(defn revert-pull [{:keys [user repo]} pull-id]
-  (g/complete-revert-pull {:body (format "Reverts %s/%s#%d" user repo pull-id)}))
+(defn revert-pull [pull-id]
+  (let [{:keys [user repo]} config]
+    (sample-pull {:title "Revert ..."
+                  :body  (format "Reverts %s/%s#%d" user repo pull-id)})))
 
 (deftest parse-issues
   (let [related (format "Related to [%s](%s)" jira-id jira-issue-url)
         fixes   (format "Fixes #1")
         body    (str/join '\n [related fixes])
-        pull    (generate-pull body)]
+        pull    (sample-pull {:body body})]
     (is (= [[jira-id jira-issue-url]
-             ["#1" "https://github.company.com/user/repo/issues/1"]] (sut/parse-issues config pull)))))
+            ["#1" "https://github.company.com/user/repo/issues/1"]] (sut/parse-issues config pull)))))
 
 (deftest parse-pull
   (testing "with a revert"
-    (are [pull-id] (= pull-id (:revert-pull (sut/parse-pull config (revert-pull config pull-id))))
+    (are [pull-id] (= pull-id (:revert-pull (sut/parse-pull config (revert-pull pull-id))))
       1
       2
       5))
   (testing "with a correct formats"
-    (are [title] (not= nil (sut/parse-pull config (g/complete-pull {:title title})))
+    (are [title] (not= nil (sut/parse-pull config (sample-pull {:title title})))
       "feat(scope): enhance this and that"
       "fix(scope): do not fail on invalid input"
       "chore: clean up the codebase"))
   (testing "with invalid formats"
-    (are [title] (nil? (sut/parse-pull config (g/complete-pull {:title title})))
+    (are [title] (nil? (sut/parse-pull config (sample-pull {:title title})))
       "this is just a PR"
       "does not follow the rules"))
   (testing "with a full test"
-    (let [pull (g/complete-pull {:title "feat(the scope): subject line" :body "Fixes #1, Closes JIRA-2"})
+    (let [pull (sample-pull {:title "feat(the scope): subject line" :body "Fixes #1, Closes JIRA-2"})
           change (sut/parse-pull config pull)]
       (is (= "feat" (:type change)))
       (is (= "the scope" (:scope change)))
       (is (= "subject line" (:subject change)))
       (is (= pull (:pull-request change)))
       (is (= 2 (count (:issues change)))))))
 
-(def pulls (map #(g/complete-valid-pull {:number %}) (range 1 5)))
-
 (defn revert [pulls]
   (->> pulls
        (map :number)
-       (map #(assoc (revert-pull config %) :number (* 10 %)))))
+       (map #(assoc (revert-pull %) :number (* 10 %)))))
+
+(defn- sample-tag [pulls]
+  (-> (spec/sample ::core-spec/tag)
+      (assoc :pulls pulls)))
```

Comment:
I would remove this threading.

Reply:
It was more readable for me this way and I was playing it a lot since generator overrides aren't always working for some reason.